src/v/cluster/topic_table.cc

```diff
@@ -384,6 +397,7 @@ topic_table::apply(cancel_moving_partition_replicas_cmd cmd, model::offset o) {
     }
     switch (in_progress_it->second.state) {
     case in_progress_state::update_requested:
+        _probe.handle_update_removal(cmd.key);
```

Comment:
This feels a bit suspicious to me: in case of cancellation we have the following sequence of events: add to `_update_in_progress`; cancel (swapping new and old replica sets); finish update. So looks like we are going to do remove twice. I think what we need instead is a function to handle update cancel - that will basically add the opposite movement.

Example: (we are on node 1, previous replica set is [1,2,3])
* add move to [2,3,4] (increment moving_from)
* cancel move (previous set becomes [2,3,4], current becomes [1,2,3] again) (decrement moving_from, increment moving_to)
* finish move (decrement moving_to)

Alternatively, we can just skip modifying counts at the cancel step, but this requires additional handling during the finish move step (we should check if we cancelled previously)

Reply:
We don't swap new and old replicas on swap. We just replace current with old